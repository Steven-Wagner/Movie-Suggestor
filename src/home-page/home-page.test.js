import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Homepage from './home-page';

describe('<Homepage/>', () => {
    it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
        <Router>
            <Homepage match={{params: {user: "1"}}}/>
        </Router>,
        div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
        .create(
            <Router>
                <Homepage match={{params: {user: "1"}}}/>
            </Router>)
        .toJSON();
        expect(tree).toMatchSnapshot();  
    });
});