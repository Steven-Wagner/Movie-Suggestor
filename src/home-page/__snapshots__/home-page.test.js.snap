// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Homepage/> renders the UI as expected 1`] = `
<div>
  <div
    className="nav-wrapper"
  >
    <nav>
      <a
        href="/friendsuggestions/1"
        onClick={[Function]}
      >
        <button
          className="nav-button remote-button"
        >
          Friend Suggestions
        </button>
      </a>
      <a
        href="/newreview/1"
        onClick={[Function]}
      >
        <button
          className="nav-button remote-button"
        >
          Review Movie
        </button>
      </a>
      <a
        href="/"
        onClick={[Function]}
      >
        <button
          className="nav-button remote-button"
          onClick={[Function]}
        >
          Logout
        </button>
      </a>
    </nav>
  </div>
  <main
    role="main"
  >
    <header
      className="main-banner"
      role="banner"
    >
      <h1>
        <span
          className="fas fa-film"
        />
        Movie Suggestor
      </h1>
      <h2
        className="subtitle"
      >
        Personalized Movie Suggestions
      </h2>
    </header>
    <div
      className="suggestions-list remote-edge"
    >
      <div
        className="error-message"
      />
      
      <p
        className="loading-message"
      >
        <span
          className="fas fa-film film-logo"
        />
        Loading...
      </p>
      
    </div>
  </main>
  <footer
    className="remote-edge"
  >
    Built by Steven Wagner. Movie data from 
    <a
      href="OMDB.com"
      target="_blank"
    >
      omdbapi.com
    </a>
    . 2019
  </footer>
</div>
`;
